clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/SubmitTitusJob.java

```diff
@@ -0,0 +1,191 @@
+   * references inside of the lambda. This should really be refactored so that pattern isn't
+   * necessary. It's really gross as-is.
+   */
+  @NotNull
+  @Override
+  public Result apply(@NotNull SubmitTitusJobCommand command, @NotNull Saga saga) {
+    final TitusDeployDescription description = command.description;
+
+    final TitusClient titusClient =
+        titusClientProvider.getTitusClient(description.getCredentials(), description.getRegion());
+
+    final SubmitJobRequest submitJobRequest = command.getSubmitJobRequest();
+    final String[] nextServerGroupName = {command.getNextServerGroupName()};
+
+    final int[] retryCount = {0};
+    String jobUri =
+        retrySupport.retry(
+            () -> {
+              try {
+                return titusClient.submitJob(submitJobRequest);
+              } catch (StatusRuntimeException e) {
+                if (isServiceExceptionRetryable(description, e)) {
+                  String statusDescription = e.getStatus().getDescription();
+                  if (statusDescription != null
+                      && statusDescription.contains(
+                          "Job sequence id reserved by another pending job")) {
+                    try {
+                      Thread.sleep(1000 ^ (int) Math.round(Math.pow(2, retryCount[0])));
+                    } catch (InterruptedException ex) {
+                      // TODO(rz): I feel like this is really bad to do...?
+                      // Sweep this under the rug...
+                    }
+                    retryCount[0]++;
+                  }
+                  nextServerGroupName[0] =
+                      TitusJobNameResolver.resolveJobName(
+                          titusClient, description, submitJobRequest);
+
+                  saga.log("Resolved server group name to '%s'", nextServerGroupName[0]);
+
+                  saga.log(
+                      "Retrying with %s after %s attempts", nextServerGroupName[0], retryCount[0]);
+                  throw e;
+                }
+                if (isStatusCodeRetryable(e.getStatus().getCode())) {
+                  retryCount[0]++;
+                  saga.log("Retrying after %s attempts", retryCount[0]);
+                  throw e;
+                } else {
+                  log.error(
+                      "Could not submit job and not retrying for status {}", e.getStatus(), e);
+                  saga.log("Could not submit job %s: %s", e.getStatus(), e.getMessage());
+                  throw e;
+                }
+              }
+            },
+            8,
+            100,
+            true);
+
+    if (jobUri == null) {
+      throw new TitusException("could not create job");
+    }
+
+    saga.log("Successfully submitted job request to Titus (Job URI: %s)", jobUri);
+
+    return new Result(
+        new ManyCommands(
+            new AttachTitusServiceLoadBalancersCommand(
```

Comment:
Even though its a noop, can we avoid running this command entirely for batch job operations?

Reply:
Yes, we do via the condition in the `SagaFlow`:

```java
SagaFlow flow = new SagaFlow()
        .then(LoadFront50App.class)
        .then(PrepareTitusDeploy.class)
        .then(SubmitTitusJob.class)
        .on(
          TitusServiceJobPredicate.class,                       // <-- HERE!
          sagaFlow -> {
            sagaFlow
              .then(AttachTitusServiceLoadBalancers.class)
              .then(CopyTitusServiceScalingPolicies.class);
          })
        .completionHandler(TitusDeployCompletionHandler.class);
```

Reply:
Lovely, thanks for sharing the code snippet here!